### PR TITLE
InputField component

### PR DESCRIPTION
**Pull request by marcus-ong-qy**
Modular Input Field component

**Active:**
Screenshot 2022-01-13 191812

**Inactive:**
Screenshot 2022-01-13 191724

### CODE REVIEW

**Comment by marcus-ong-qy** [comment]
[input field](https://trello.com/c/rHdIHP8p/12-input-field)

File: src/components/InputField/styles/InputField.ts [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+  width: 100%;
+`
+
+export const InputFieldTitle = styled.p.attrs({ colour: 'white' })`
+  margin: 0px;
+
+  font-style: normal;
+  font-weight: 500;
+  font-size: 18px;
+  line-height: 50px;
```
Comment (hui444) [inline src/components/InputField/styles/InputField.ts:13]: any reason line height is way bigger than 18px? consider using values like 1 or 1.2?

File: src/components/InputField/styles/InputField.ts [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+export const TextInput = styled.input.attrs({ borderColorInactive: 'gray', borderColor: 'white' })`
+  width: 100%;
+
+  outline: 0;
+  border: 0;
+  border-bottom: 1px solid ${(props) => props.borderColorInactive};
+
+  font-weight: 500;
+  font-size: 18px;
+  line-height: 50px;
```
Comment (hui444) [inline src/components/InputField/styles/InputField.ts:27]: this too, if its intentional then seems ok but might be a little odd,. but do consider making font size responsive using `clamp()`

**Review by rldefa — COMMENTED** [review]
Seems good to me, clamp does sound better.

File: src/components/InputField/styles/InputField.ts [hunk 3]
```diff
@@ -0,0 +1,37 @@
+import styled from 'styled-components'
+
+export const InputFieldStyled = styled.div`
+  width: 100%;
+`
+
+export const InputFieldTitle = styled.p.attrs({ colour: 'white' })`
+  margin: 0px;
```
[the remark this answers is not in the document]
Comment (marcus-ong-qy) [reply src/components/InputField/styles/InputField.ts]: fixed in 8a9e69fb99032cf3ebc79e5502d5b6a086a45f1e

File: src/components/InputField/styles/InputField.ts (hunk 2 above)
Comment (marcus-ong-qy) [reply src/components/InputField/styles/InputField.ts]: fixed in 8a9e69fb99032cf3ebc79e5502d5b6a086a45f1e, organised font params in theme file

File: src/components/InputField/InputField.tsx [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+import { InputFieldStyled, InputFieldTitle, TextInput } from './styles/InputField'
+
+// TODO add error handling
+function InputField({ title, value, updateValue }: { title: string; value?: string; updateValue?: any }) {
+  const theme = useTheme()
+  const palette = { ...theme.palette }
+
+  return (
+    <InputFieldStyled>
+      <InputFieldTitle colour={palette.common.white}>{title}&thinsp;:</InputFieldTitle>
```
[the remark this answers is not in the document]
Comment (marcus-ong-qy) [reply src/components/InputField/InputField.tsx]: fixed in 8a9e69fb99032cf3ebc79e5502d5b6a086a45f1e, replaced with inline padding

File: src/components/InputField/styles/InputField.ts (hunk 1 above)
Comment (marcus-ong-qy) [reply src/components/InputField/styles/InputField.ts]: followed styling based on figma file; replaced line-height with relative sizing in 8a9e69fb99032cf3ebc79e5502d5b6a086a45f1e

**Comment by marcus-ong-qy** [comment]
Organised the font styling into the theme file in b9606819e03248c6b73049c26aac813437742119

Pass in the FontType data type into the styled component, then the fontTypeCss css component imported from index.styled will take the fontType prop and set the css accordingly.

File: src/styles/Theme.tsx [hunk 5]
```diff
@@ -6,12 +6,43 @@ const theme: DefaultTheme = {
...
       white: '#fff',
+      gray: '#888',
     },
     primary: '#10bcbc',
     secondary: '#5e2ba1',
   },
   typography: {
     main: 'Inter',
+    fontSize: {
+      main: {
```
Comment (hui444) [inline src/styles/Theme.tsx:17]: consider using 'body' for normal text size?

**Review by hui444 — APPROVED** [review]
lgtm but do remove package-lock file!!!!

**Pull request merged**